Fix behaviour of child answer checkboxes on 0.0.1 payloads

kierenbeckett opened the pull request:
### What is the context of this PR?
In 0.0.1 payloads the data should contain the value of the child answer rather than the main value (e.g. 'other')

### How to review 
Run `test_checkbox.json`, tick Other and enter a child answer, look at `/dump/submission`. Previous behaviour:

```
    "data": {
      "1": "Cheese", 
      "20": "['Ham', 'Pineapple', 'Other']", 
      "4": "Tuna", 
      "6": "Other"
    }, 
```

New behaviour:

```
    "data": {
      "1": "Cheese", 
      "20": "['Ham', 'Pineapple', 'enteredvalue2']", 
      "4": "Tuna", 
      "6": "enteredvalue1"
    }, 
```

If Other is ticked but no child answer is entered:

```
    "data": {
      "1": "Cheese", 
      "20": "['Ham', 'Pineapple', 'Other']", 
      "4": "Tuna", 
      "6": "Other"
    }, 
```

jonnyshaw89 reviewed (approved):
👍 This looks good. Nice work

nealedj commented on data/en/test_checkbox.json:
```diff
@@ -165,4 +165,4 @@
     }]
-}
\ No newline at end of file
+}
```
Could you remove the new line at the end of the file? It's the default behaviour of js-beautify so when we run `yarn format` it'll remove the new line.

nealedj reviewed (approved):
Looks good - nice tidy up too. 👍

Could you remove the new line on the schema and then squash please?

kierenbeckett committed:
Fix behaviour of child answer checkboxes on 0.0.1 payloads

In 0.0.1 payloads the data should contain the value of the child
answer rather than the main value e.g. 'other'
